![Image](https://parentinfo.org/sites/default/files/styles/main_article_image/public/Social_Media_image_New_IG_Logo.png?itok=jDLmx9B5)
# The Most Popular App Genres

**Android** and **iOS** mobile apps are developed and offered to a variaty of users. Apps provided for free have as its main source of revenue in-app adds. The objective of this analysis is to *verify the type of apps with highest rates of downloads and usage*.

We used two datasets available online for this analysis:
- A [data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately 10,000 Android apps from Google Play. The data was collected in August 2018. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

- A [data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately 7,000 iOS apps from the App Store. The data was collected in July 2017. You can download the data set directly from this [link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

## Opening and cleaning the data

In [1]:
# Open dataset with Android apps
open_android = open ('googleplaystore.csv')
from csv import reader
read_android = reader (open_android)
android = list (read_android)
android_header = android[0]
android = android[1:]

# Open dataset with IOS apps
open_ios = open ('AppleStore.csv')
from csv import reader
read_ios = reader (open_ios)
ios = list (read_ios)
ios_header = ios[0]
ios = ios[1:]

In [2]:
# Define a function to help us explore the datasets
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
# Explore the android dataset
print(android_header)
print('\n')
explore_data(android, 0, 3, True )

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [4]:
# Explore the ios dataset
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


In [5]:
# Print the column names of the Android and iOS datasets
print ('Android Dataset Column Names')
print (android_header)
print ('\n') # Print an empty row
print ('iOS Dataset Column Names')
print (ios_header)

Android Dataset Column Names
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


iOS Dataset Column Names
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The Google Play dataset has 13 columns including the App name, the Category and the Rating. The iOS dataset comprises 16 columns with similar information if compared to the Android dataset.

### Removing wrong data
While exploring the datasets, we see that the Android file includes a row with missing columns.

In [6]:
# Check on row 10473 that seems to have an error: category column is missing
android[10472]

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

In [7]:
# Remove row with error
del android[10472]

### Dropping duplicate data

We also observe that the Google Play dataset (Android) includes duplicate entries. In the code below we show multiple rows for Instagram and Facebook.

In [8]:
# Print all rows with the app name Facebook
for app in android:
    name = app[0]
    if name == 'Facebook':
        print (app)
        print ('\n') # Print an empty row between each output row

['Facebook', 'SOCIAL', '4.1', '78158306', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']


['Facebook', 'SOCIAL', '4.1', '78128208', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'August 3, 2018', 'Varies with device', 'Varies with device']




In [9]:
# Print all rows with the app name Instagram
for app in android:
    name = app[0]
    if name == 'Instagram':
        print (app)
        print ('\n') # Print an empty row between each output row

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']




In [10]:
# Count the number of duplicate rows
duplicate_names = []
unique_names = []

for app in android:
    name = app[0]
    
    if name in unique_names:
        duplicate_names.append(name)
    else:
        unique_names.append(name)
        
print (len(duplicate_names))

1181


In the code above, we see that the Google Play dataset includes 1,181 duplicate rows. The next step in our analysis is to remove the duplicates. The criterion we used to chose one of the rows per app is the one with the highest number of reviews. The logic behind is that we expect the entries with higher number of reviews to be the most recent.

In [11]:
# Create a dictionary only with unique apps as keys and their highest reviews as values
reviews_max = {}

for app in android: # Disregard the dataset header
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews     

In [12]:
# Display the number of unique apps and their maximum number of reviews
len(reviews_max)

9659

In the code below we:
1. Create two empty lists to store our new clean data set and keep track of the app names
2. Loop through the Google Play dataset selecting the rows with the highest reviews that are not yet in the list
3. Append these rows to the new clean list

In [13]:
# Remove the duplicate rows
android_clean = [] # Create an empty list where we will create our clean dataset
already_added = [] # Create an empty list where we will keep track of the names of the added apps

for app in android: 
    name = app[0] # Create a variable with the name of the app
    n_reviews = float(app[3])
    
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)

After removing the duplicate data, the clean dataset has 9,549 rows.

In [14]:
# Explore the android_clean dataset, it is expected to have 9659 rows
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


## Removing apps written in languages different than English

Both datasets include apps written in other languages. We will clean the data to include only the ones written in English.

In [15]:
# Create a function that returns False if non-English characters are included in a string
def is_english (string):
    
    for character in string:
        if ord(character) > 127:
            return False
        
    return True

In [16]:
# Testing the function: a non-English character should return False and an English one True
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))
print (ord('™'), ord('😜'))

True
False
False
False
8482 128540


As we see in the code above, '™' and '😜' are characters out of the range of typical English characters. That's why the code returns False to the last two apps that contain these characters. As a way to minimize this, we will edit the previous function and make it return False only if more than 3 characters are classified as non-English.

In [17]:
# Edit the function to return False only to apps that include more than 3 characters classified as non-English
def is_english (string):
    tot_non_engl = 0
    for character in string:
        if ord(character) > 127:
            tot_non_engl += 1
            if tot_non_engl > 3:
                return False
        
    return True

In [18]:
# Check if the function is working properly with diffeernt statements
print(is_english ('Docs To Go™ Free Office Suite'))
print(is_english ('Instachat 😜'))
print(is_english ('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
True
False


In the code above se see that now the function returns False only to non-English terms.

In [19]:
# Filter out non-English apps from the android_clean dataset
android_english = []
for app in android_clean:
    name = app[0]
    if is_english (name):
        android_english.append(app)

In [20]:
# Explore the new android_english dataset without the non-English rows
explore_data(android_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


In [21]:
# Filter out non-English apps from the ios datset
ios_english = []
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)

In [22]:
# Explore the new ios_english dataset without the non-English rows
explore_data(ios_english, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns: 16


After removing the non-English apps, the android dataset has now 9,614 rows and the iOS dataset 6,183 entries.

## Removing non-free apps
The final step in cleaning our data is to remove the apps that are not free.

In [23]:
# Check the headers of both datasets to see where the free/non-free information is
print(android_header)
print('\n')
print(ios_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


The free apps have the value '0' in column 8 of the Android dataset and in column 5 of the iOS dataset. In the following code we identify the free apps in both datasets and save them in a new list of lists.

In [24]:
# Create new datasets including only free apps
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = float(app[4])
    if price == 0:
        ios_final.append(app)

In [25]:
# Exploring final clean datasets including only free apps
print(len(android_final))
print(len(ios_final))

8864
3222


Our final clean datasets to be used in the analysis are android_final with 8,864 entries and and ios_final with 3,222.

## Data analysis
The objective of this investigation is to identify the types of apps with the highest potential in terms of usage. Let's check each dataset columns to identify which ones could provide the information we need.

In [26]:
# Checking on the first lines of both datasets
print (android_header)
print ('\n')
print (android[0])
print ('\n')
print (ios_header)
print ('\n')
print (ios[2500])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['504170504', 'Louvre HD', '125812736', 'USD', '1.99', '1129', '93', '5.0', '5.0', '4.4', '4+', 'Lifestyle', '40', '4', '12', '1']


The apps genre can be identified in the 'Genres' and 'Category' columns in the Google Play dataset and in the 'prime_genre' column in the iOS dataset.


### Creating a frequency table with the different genres

In [27]:
# Creating a freq_table function returning a frequency table for any column in a dataset
def freq_table (dataset, index):
    
    table = {}
    
    for row in dataset:
        item = row[index]
        
        if item in table:
            table[item]+=1  
        else:
            table[item]=1

    for topic in table:
        table[topic] /= len (dataset)
        table[topic] *= 100
        
    return table

In [28]:
# Defining a function that takes our freq_table function and reorders the topics
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [29]:
# Display prime_genre from ios_final
display_table (ios_final, -5) 

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Analysing the iOS dataset, we see that most apps are related to entertainment and fun. They include topics like gaming, photos and videos. Gaming is the theme that stands out the most, comprising much more apps than the other genres. The runner-up is Entertainment. 

A search on the App Store for the category 'Games' brought the results displayed in the picture below (search made on April, 11 2020). 

![ios](ios_pic.png)

In [30]:
# Display Category from android_final
display_table (android_final, 1) 

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [31]:
# Display Genres from android_final
display_table (android_final, -4) 

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

When it comes to the Google Play results, apps profiles tend to be more skewed towards practical topics as Tools, Education and Business. At the same time, Entertainment and Games still appear  at the top of the list. These last two themes stand out in both platforms.

### Calculating the average number of Reviews and Installs for different genres
The previous analysis gave us insight on the types of apps we can find in each platform. We will now look at the data from a different angle, identifying the genres with the largest number of installs and reviews. 

As the number of installs is only available at the Google Play dataset, we use the reviews as a proxi while analysing the App Store.

### Average Number of reviews in the App Store

In [32]:
# Calculate the average number of reviews by genre in the App Store dataset
unique_genres = []

for row in ios_final:
    genre = row[-5]
    if genre not in unique_genres:
        unique_genres.append(genre)

for genre in unique_genres:
    total = 0
    len_genre = 0
    
    for item in ios_final:
        genre_app = item[-5]
        if genre_app == genre:
            rating = float(item[5])
            total+=rating
            len_genre+=1   
            
    avg_rating = total / len_genre
    print(genre, ':', avg_rating)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


The two genres with the largest number of reviews in the iOS dataset are Navigation and Reference. Let's have a look at the apps included in each of them.

In [33]:
# Displaying Navigation apps
for row in ios_final:
    genre = row[-5]
    name = row[1]
    reviews = row[5]
    if genre == 'Navigation':
        print(name, ':', reviews)

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


While observing the apps comprised in the Navigation genre, we see that Waze and Google Maps are responsible for most of the reviews. It is unlikely that new apps at the same category will present similar figures. At the same time, these results show that geolocation information may be highly relevant to users.

In [34]:
# Displaying Reference apps
for row in ios_final:
    genre = row[-5]
    name = row[1]
    reviews = row[5]
    if genre == 'Reference':
        print(name, ':', reviews)

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


The Reference genre includes different types of apps, encompassing the ones related to religion, dictionaries and translators. This analysis highlights the importance of religion and the need for understanding different languages. Apps linked to such topics may be relevant to a large number of users.

### Average number of installs in Google Play

In [35]:
# Print first 3 roow of the Installs column of the android_final dataset
installs_list = []
for row in android_final:
    installs = row[5]
    installs_list.append(installs)
    
print(installs_list[:5])

['10,000+', '5,000,000+', '50,000,000+', '100,000+', '50,000+']


We see that the number of Installs is displayed in an open-ended way, not showing the precise numbers but an approximation. Besides, values include plus and comma signs, what prevents us from converting them directly into floats or integers. Therefore, we will need to remove the plus and comma signs before performing calculations with the Installs column.

In the code below, we calculate the total number of Installs per genre in the Google Play dataset. The code includes a step in which we remove the plus and comma signs from the values.

In [36]:
# Create a list with the unique genres in the Category column
unique_genres = []

for row in android_final:
    genre = row[1]
    if genre not in unique_genres:
        unique_genres.append(genre)
        
print ('List of Unique Genres: ',unique_genres)

List of Unique Genres:  ['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY', 'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION', 'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE', 'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME', 'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL', 'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL', 'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER', 'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION']


In [37]:
# Calculate the average number of Installs per genre
for genre in unique_genres:
    total = 0
    len_genre = 0
    
    for row in android_final:
        app_genre = row[1]
        if app_genre == genre:
            installs = row[5]
            installs = installs.replace(',', '') # Removing comma signs from the values
            installs = installs.replace('+', '') # Removing plus signs from the values
            installs = float(installs)
            total += installs
            len_genre += 1
            
    avg_install = total / len_genre
    print(genre, ': ', avg_install)

ART_AND_DESIGN :  1986335.0877192982
AUTO_AND_VEHICLES :  647317.8170731707
BEAUTY :  513151.88679245283
BOOKS_AND_REFERENCE :  8767811.894736841
BUSINESS :  1712290.1474201474
COMICS :  817657.2727272727
COMMUNICATION :  38456119.167247385
DATING :  854028.8303030303
EDUCATION :  1833495.145631068
ENTERTAINMENT :  11640705.88235294
EVENTS :  253542.22222222222
FINANCE :  1387692.475609756
FOOD_AND_DRINK :  1924897.7363636363
HEALTH_AND_FITNESS :  4188821.9853479853
HOUSE_AND_HOME :  1331540.5616438356
LIBRARIES_AND_DEMO :  638503.734939759
LIFESTYLE :  1437816.2687861272
GAME :  15588015.603248259
FAMILY :  3695641.8198090694
MEDICAL :  120550.61980830671
SOCIAL :  23253652.127118643
SHOPPING :  7036877.311557789
PHOTOGRAPHY :  17840110.40229885
SPORTS :  3638640.1428571427
TRAVEL_AND_LOCAL :  13984077.710144928
TOOLS :  10801391.298666667
PERSONALIZATION :  5201482.6122448975
PRODUCTIVITY :  16787331.344927534
PARENTING :  542603.6206896552
WEATHER :  5074486.197183099
VIDEO_PLAYERS 

From the information above, we see that three of the genres stand out at Google Play, with more than 20 million installs on average: Communication, Social and Video_Player. Below, we check what are the apps included in each of these groups.

In [38]:
# Print the app names and number of installs for the Communication Category
for row in android_final:
    name = row[0]
    category = row[1]
    installs = row[5]
    if category == 'COMMUNICATION':
        print(name, ':', installs)

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

In the Communication category, WhatsApp alone has more than 1 billion installs. At the same time, se can see other apps from the same category with also a substantial number of installs (e.g. imo beta free calls and text). This seems to be a genre with potential to attract a large number of users.

In [39]:
# Print apps from the Social category.
# Print the app names and number of installs for the Communication Category
for row in android_final:
    name = row[0]
    category = row[1]
    installs = row[5]
    if category == 'SOCIAL':
        print(name, ':', installs)

Facebook : 1,000,000,000+
Facebook Lite : 500,000,000+
Tumblr : 100,000,000+
Social network all in one 2018 : 100,000+
Pinterest : 100,000,000+
TextNow - free text + calls : 10,000,000+
Google+ : 1,000,000,000+
The Messenger App : 1,000,000+
Messenger Pro : 1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus : 1,000,000+
Telegram X : 5,000,000+
The Video Messenger App : 100,000+
Jodel - The Hyperlocal App : 1,000,000+
Hide Something - Photo, Video : 5,000,000+
Love Sticker : 1,000,000+
Web Browser & Fast Explorer : 5,000,000+
LiveMe - Video chat, new friends, and make money : 10,000,000+
VidStatus app - Status Videos & Status Downloader : 5,000,000+
Love Images : 1,000,000+
Web Browser ( Fast & Secure Web Explorer) : 500,000+
SPARK - Live random video chat & meet new people : 5,000,000+
Golden telegram : 50,000+
Facebook Local : 1,000,000+
Meet – Talk to Strangers Using Random Video Chat : 5,000,000+
MobilePatrol Public Safety App : 1,000,000+
💘 WhatsLov: Smileys of love, sti

In the Social category, Facebook and Facebook Lite sum up more than 1.5 billion installs. We have also other apps with more than 100 million (e.g. Pinterest). Not many apps in this category reach a large number of users though. That seems to be restricted to a small number of well-known apps.

In [40]:
# Print apps from the Social category.
# Print the app names and number of installs for the Communication Category
for row in android_final:
    name = row[0]
    category = row[1]
    installs = row[5]
    if category == 'VIDEO_PLAYERS':
        print(name, ':', installs)

YouTube : 1,000,000,000+
All Video Downloader 2018 : 1,000,000+
Video Downloader : 10,000,000+
HD Video Player : 1,000,000+
Iqiyi (for tablet) : 1,000,000+
Video Player All Format : 10,000,000+
Motorola Gallery : 100,000,000+
Free TV series : 100,000+
Video Player All Format for Android : 500,000+
VLC for Android : 100,000,000+
Code : 10,000,000+
Vote for : 50,000,000+
XX HD Video downloader-Free Video Downloader : 1,000,000+
OBJECTIVE : 1,000,000+
Music - Mp3 Player : 10,000,000+
HD Movie Video Player : 1,000,000+
YouCut - Video Editor & Video Maker, No Watermark : 5,000,000+
Video Editor,Crop Video,Movie Video,Music,Effects : 1,000,000+
YouTube Studio : 10,000,000+
video player for android : 10,000,000+
Vigo Video : 50,000,000+
Google Play Movies & TV : 1,000,000,000+
HTC Service － DLNA : 10,000,000+
VPlayer : 1,000,000+
MiniMovie - Free Video and Slideshow Editor : 50,000,000+
Samsung Video Library : 50,000,000+
OnePlus Gallery : 1,000,000+
LIKE – Magic Video Maker & Community : 50,

In the Video Player category, YouTube drives the average upwards, with more than 1 billion installs. Still, we can find many other apps in the same category displaying a large number of installs, in some cases surpassing 500 million.

# Conclusions

The analysis offers multiple avenues of opportunity in terms of genres with the potential to attract a large number of users. 

The data show that there seems to be a high amoung of users interested in subjects related to religion. They seek ways of being able to read the holy books and spiritual content. Apps that provide this type of information seem to have potential.

Another topic that tends to be highly relevant among users is the understanding of different languages. Dictionaries and translators attract a wide array of users. These types of services also show encouraging results.

Especially in big cities, users have the need to swiftly communicate and access geolocation information of people and places. Apps that can help users finding the right places and connecting with people in the area they live seem to be highly relevant as well.

Finally, users tend to appreciate platforms that allow them to interact with video content. Therefore, apps that can convey a way for users to learn and get informed interacting with video seem also to have the potential of atracting a high number os users.